dbms/src/Functions/FunctionsRegexpReplace.h

```diff
@@ -220,7 +224,7 @@ class FunctionStringRegexpReplace : public FunctionStringRegexpBase
             std::unique_ptr<Regexps::Regexp> regexp;
             if (col_size > 0)
             {
-                regexp = memorize<true>(pat_param, match_type_param, collator);
+                regexp = memorize<false>(pat_param, match_type_param, collator, replace_default_flag);
```

Comment:
why change `need_subpattern` to false here?

Reply:
> why change `need_subpattern` to false here?

At the beginning, we use `FindAndConsume` to implement the `regexp_replace`, true is needed to add pattern. And now, we implement the `regexp_replace` with `Match`, so the added pattern is no more need.